fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java

```diff
@@ -411,20 +411,84 @@ private StringBuilder createSearchQuery(final SearchParameters parameters,
         final var fields = new ArrayList<String>(queryFields);
         final var rdfTypeConditionValue =
                 conditions.stream().filter(c -> c.getField().equals(RDF_TYPE)).findFirst().orElse(null);
-        final var containsRDFTypeField = queryFields.contains(RDF_TYPE.toString());
 
-        final var groupByFields = fields.stream().filter(f->!f.equals(RDF_TYPE.toString())).collect(toList());
-        final var returnFields = new ArrayList<>(groupByFields);
+        if (isPostgres()) {
+            return createPostgresOptimizedQuery(parameterSource, whereClauses, conditions, fields,
+                    rdfTypeConditionValue);
+        } else {
+            return createDefaultQuery(parameterSource, whereClauses, conditions, fields, rdfTypeConditionValue);
+        }
```

Comment:
Instead of `return`ing right from here, could you log the SQL returned from either of the two functions above. This could be at either `DEBUG` or `TRACE` level. It's really hard to tell what query is being generated here. This would help us see the result now and in future.

Reply:
Good idea.

Reply:
Added.